server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/EnvironmentController.java

```diff
@@ -53,4 +59,28 @@ public SuccessResponse<Environment> getEnvironment(@PathVariable Long id) {
         return Responses.success(environmentService.detail(id));
     }
 
+    @ApiOperation(value = "createEnvironment", notes = "Create an environment")
+    @RequestMapping(value = "/environments", method = RequestMethod.POST)
+    public SuccessResponse<Environment> createEnvironment(@RequestBody @Valid CreateEnvironmentReq req) {
```

Comment:
Should add `@Validated` for the controller class?

Reply:
Just adding `@Validated` on the class level will not help you validate all the methods. You still need to add `@Valid` or `@Validated` in the front of method parameters.